l1-contracts/src/core/Decoder.sol

```diff
@@ -115,15 +121,16 @@ contract Decoder {
       calldatacopy(add(temp, 0x20), add(_l2Block.offset, 0x04), size)
     }
 
-    // Diff root
-    (bytes32 diffRoot, bytes32 l1ToL2messagesHash) = _computeDiffRootAndMessagesHash(_l2Block);
+    bytes32 diffRoot;
+    bytes32 l1ToL2messagesHash;
+    (diffRoot, l2ToL1Msgs, l1ToL2Msgs, l1ToL2messagesHash) = _computeConsumables(_l2Block);
     assembly {
       let endOfTreesData := sub(0x23c, 0x04)
       mstore(add(temp, add(0x20, endOfTreesData)), diffRoot)
       mstore(add(temp, add(0x40, endOfTreesData)), l1ToL2messagesHash)
     }
 
-    return bytes32(uint256(sha256(temp)) % P);
+    return (bytes32(uint256(sha256(temp)) % P), l2ToL1Msgs, l1ToL2Msgs);
```

Comment:
may as well make the mod p variable named on its own then you can spare the return statement. Having a calc in the return statement while others are pre calculated is a bit messy can we be consistent?